packages/snaps-utils/src/handlers.ts

```diff
@@ -194,7 +194,7 @@ export type OnNameLookupResponse =
 
 export type OnNameLookupArgs = {
   chainId: Caip2ChainId;
-} & ({ domain: string } | { address: string });
+} & ({ domain: string; address?: never } | { address: string; domain?: never });
```

Comment:
Do we need the same on `OnNameLookupResponse` as that also uses a union to prevent both values at the same time?

Reply:
I'm not sure if it makes sense to do this for the response. This essentially allows for destructuring, correct? It wouldn't make sense to do that given the response could be null. I think I also want to force us to use a type guard to narrow the type where ever we use this (in the extension).